### PR TITLE
Make `seq` return a `ListStream` where possible.

### PR DESCRIPTION
**Pull request by merelymyself**
# Description

Significant perf improvements as well: try something like `seq 1 100000000000` and observe how it is streaming correctly.

# Tests

Make sure you've done the following:

- [ ] Add tests that cover your changes, either in the command examples, the crate/tests folder, or in the /tests folder.
- [ ] Try to think about corner cases and various ways how your changes could break. Cover them with tests.
- [x] If adding tests is not possible, please document in the PR body a minimal example with steps on how to reproduce so one can verify your change works.

Make sure you've run and fixed any issues with these commands:

- [ ] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [ ] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [ ] `cargo test --workspace --features=extra` to check that all the tests pass

# Documentation

- [ ] If your PR touches a user-facing nushell feature then make sure that there is an entry in the documentation (https://github.com/nushell/nushell.github.io) for the feature, and update it if necessary.

### CODE REVIEW

File: crates/nu-command/src/generators/seq.rs [hunk 1]
```diff
@@ -324,74 +326,87 @@ fn print_seq(
...
     // for string output
     let mut ret_str = "".to_owned();
-    // for number output
-    let mut ret_num = vec![];
     // If the separator and terminator are line endings we can convert to numbers
     let use_num =
         (separator == "\n" || separator == "\r") && (terminator == "\n" || terminator == "\r");
 
+    if use_num {
+        let int_stream: bool = first.fract() == 0.0 && step.fract() == 0.0;
```
Comment (sholderbach) [inline crates/nu-command/src/generators/seq.rs:340]: I don't want to trust float precision for any precise counting. We should specialize the implementation to either use a `Counter<i64>` or `Counter<f64>`.

File: crates/nu-command/src/generators/seq.rs [hunk 2]
```diff
@@ -324,74 +326,87 @@ fn print_seq(
...
-    } else {
-        let rows: String = ret_str.lines().collect();
-        Value::string(rows, span).into_pipeline_data()
-    }
+struct Counter {
+    count: f64,
+    step: f64,
+    last: f64,
+    span: Span,
+    int: bool,
```
Comment (sholderbach) [inline crates/nu-command/src/generators/seq.rs:387]: I think this can be generic.

**Review by sholderbach — CHANGES_REQUESTED** [review]
That sounds like a worthwhile improvement, especially as in most cases you don't actually care about the small value being emitted.

But I think for correctness reasons, we should change the implementation to use the most appropriate numeric type. Ints when the bounds are ints and floats if float intervals are requested.

File: crates/nu-command/src/generators/seq.rs [hunk 3]
```diff
@@ -324,74 +357,110 @@ fn print_seq(
     pad: bool,
     padding: usize,
     span: Span,
+    is_float_sequence: bool,
```
Comment (sholderbach) [inline crates/nu-command/src/generators/seq.rs:360]: Mhh, the data is still coming in as float and may have lost precision. Could we just pass in the `Counter`?

**Comment by fdncred** [comment]
Hey @merelymyself, do you have time to continue working on this one? Seems like you've been busy recently and haven't got back to it.

**Comment by sholderbach** [comment]
This has been kind of overtaken by #7045, but should now be easier to make consistently lazy.

**Comment by fdncred** [comment]
> Hello! Sorry for the long period of inactivity...

Hey! It's @merelymyself!!! It's so glad to see you back here. Welcome again!

**Pull request closed without merging**